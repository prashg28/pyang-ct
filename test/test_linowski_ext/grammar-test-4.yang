module grammar-test {

   namespace "urn:org:example:xml:ct:grammar-test";
   prefix "gr";

   import complex-types { prefix "ct"; }

   ct:complex-type ct1 {
   }

   extension ext1;

   feature f1;

   gr:ext1 {}

   ct:complex-type ct2 {
       
       ct:abstract false {
           gr:ext1;
           // error: grammar error
           ct:abstract true;
       }

       ct:extends gr:ct1 {
           description "desc";
           reference "ref";
           status "current";

           // errors: double definitions
           description "desc";
           reference "ref";
           status "current";
       }

       ct:instance ct1 {
           description "desc";
           config true;
           if-feature f1;
           mandatory true;
           must "";
           reference ref;
           status current;
           ct:instance-type ct1;
           when "";

           // double definition
           // error:
           description "desc";
           // error:
           config true;
           if-feature f1;
           // error:
           mandatory true;
           must "";
           // error:
           reference ref;
           // error:
           status current;
           // error:
           type ct1;
           // error:
           when "";
       }

       ct:instance-list ct2 {
           description "desc";
           config true;
           if-feature f1;
           // error: mandatory is not allowed
           mandatory true;
           must "";
           reference ref;
           status current;
           ct:instance-type ct2;
           when "";
           min-elements 1;
           max-elements 2;

           // double definition
           // error:
           description "desc";
           // error:
           config true;
           if-feature f1;
           // error:
           mandatory true;
           must "";
           // error:
           reference ref;
           // error:
           status current;
           // error:
           type ct1;
           // error:
           when "";
           min-elements 1;
           max-elements 2;

          
           
       }



   }


}
