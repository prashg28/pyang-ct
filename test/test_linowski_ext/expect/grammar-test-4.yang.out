grammar-test-4.yang:22: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:31: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:32: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:33: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:49: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:51: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:54: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:57: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:59: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:61: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:63: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:66: error: KEY_REQUIRED
grammar-test-4.yang:71: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:82: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:84: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:87: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:90: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:92: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:94: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:96: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:97: error: UNEXPECTED_KEYWORD
grammar-test-4.yang:98: error: UNEXPECTED_KEYWORD
